sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -295,6 +295,70 @@ export class ApplicationOfferingChangeRequestService {
     });
   }
 
+  /**
+   * Get the Application Offering Change Request Status by the application id.
+   * @param applicationId the application id.
+   * @param studentId the student id.
+   * @returns application offering change request status.
+   */
+  async getApplicationOfferingChangeRequestStatus(
+    applicationId: number,
+    studentId: number,
+  ): Promise<ApplicationOfferingChangeRequestStatus> {
+    const applicationOfferingChangeRequest =
+      await this.applicationOfferingChangeRequestRepo.findOne({
+        select: {
+          id: true,
+          applicationOfferingChangeRequestStatus: true,
+          createdAt: true,
+        },
+        relations: {
+          application: {
+            student: true,
+          },
+        },
+        where: {
+          application: { id: applicationId, student: { id: studentId } },
+        },
+        order: {
+          createdAt: "DESC",
+        },
+      });
+    return applicationOfferingChangeRequest?.applicationOfferingChangeRequestStatus;
+  }
+
+  /**
+   * Get all the Application Offering Change Requests that are in the in progress or declined state for the provided application id.
+   * @param applicationId the application id.
+   * @param studentId the student id.
+   * @param applicationOfferingChangeRequestStatuses list of application offering change request statuses.
+   * @returns application offering change requests.
+   */
+  async getApplicationOfferingChangeRequestsByStatus(
+    applicationId: number,
+    studentId: number,
+    applicationOfferingChangeRequestStatuses: ApplicationOfferingChangeRequestStatus[],
+  ): Promise<ApplicationOfferingChangeRequest[]> {
+    return this.applicationOfferingChangeRequestRepo.find({
+      select: {
+        id: true,
+        applicationOfferingChangeRequestStatus: true,
+        createdAt: true,
+      },
+      relations: {
```

Comment:
just double checking, do we need `relations` just for the sake of `where`. I may be wrong, but I remember a comment in a PR, stating something like that, can you double-check?

Reply:
Thank You @ann-aot. I removed the `relations`. It is not needed in this scenario.

Reply:
We do not need relations for the `where`.